crates/http-source/src/main.rs

```diff
@@ -19,17 +24,64 @@ const SIGNATURES: &str = concat!("InfinyOn HTTP Source Connector ", env!("CARGO_
 #[connector(source)]
 async fn start(config: HttpConfig, producer: TopicProducer) -> Result<()> {
     debug!(?config);
+
     let source = HttpSource::new(&config)?;
-    let mut stream = source.connect(None).await?;
+    let initial_response = source.issue_request().await?;
 
-    info!("Starting {SIGNATURES}");
+    if is_chunked_transfer_response(initial_response.headers())? {
```

Comment:
This is a leaking of low-level details and abstractions to high-level `start` function. 
Here, we have a config, we construct something that implements `Source` and we read from it. That's it. The details of following the contract for `Source` trait must be hidden inside `HttpSource` struct.